microsoft/testsuites/vm_extensions/scripts/cvt.sh

```diff
@@ -0,0 +1,204 @@
+
+    if [[ $# -ne 4 ]]; then
+        echo "Insufficient number of args"
+        return 1
+    fi
+
+    local source_dev, target_dir, subtestname, testname
+    local workingdir, cvt_logs_dir, cvt_log_file, cvt_op_file
+    local lsmodop, grepop
+    source_dev=$1
+    target_dir=$2
+    subtestname=$3
+    testname=$4
+    workingdir=$PWD
+    cvt_logs_dir="$workingdir/cvt_logs"
+    cvt_log_file="$cvt_logs_dir/cvtlog_$subtestname.txt"
+    cvt_op_file="$cvt_logs_dir/cvt_$subtestname.txt"
+
+
+    lsmodop=$(lsmod | grep "involflt")
+    if [[ -z "${lsmodop// }" ]]; then
+        echo "Driver not loaded, exiting"
+        return 1
+    fi
+
+    if [[ ! -e $target_dir ]]; then
+        echo "Target directory $target_dir doesn't exist"
+        return 1
+    fi
+
+    if [[ ! -e $source_dev ]]; then
+        echo "Source dev $source_dev doesn't exist"
+        return 1
+    fi
+
+    if [ "$testname" = "ditest" ]; then
+        sourcedevsize=$(blockdev --getsize64 "$source_dev")
+        if [[ "$sourcedevsize" -gt 2147483648 ]]; then
+            echo "Source dev $source_dev size is $sourcedevsize"
+            echo "CVT works on source dev of size <= 2GB"
+            return 1
+        fi
+    fi
+
+    /usr/local/ASR/Vx/bin/stop
+    execute_cmd "mkdir -p $cvt_logs_dir"
+
+    chmod +x "$workingdir/indskflt_ct"
+    /usr/local/ASR/Vx/bin/inm_dmit --op=start_notify&
+    execute_cmd "time $workingdir/indskflt_ct --tc=$testname --loggerPath=$cvt_logs_dir --pair[ -type=d-f -sd=$source_dev -td=$target_dir/target_file.tgt -subtest=$subtestname -log=$cvt_log_file ] >> $cvt_op_file 2>&1" "ignorefail"
+    killall inm_dmit
+    pkill inm_dmit
+
+    grepop=$(grep -i "DI Test Passed" < "$cvt_op_file")
+    if [[ -z "${grepop// }" ]]; then
+            echo "CVT test failed"
+            mv "$cvt_log_file" "$cvt_log_file"."$(date)"
+            mv "$cvt_op_file" "$cvt_op_file"."$(date)"
+            return "$FAILED_TEST"
+    else
+            echo "CVT test passed"
+            return "$PASSED_TEST"
+    fi
+
+}
+
+run_tests()
+{
+    local failed, stime, ctests, ntests
+    local diskname, testcases, curdir
+    failed=0
+    stime=10
+    ctests=0
+
+    umount /data
+    diskname=$(fdisk -l 2>/dev/null | grep -o "/dev/sd[d-i]")
+    log "Formatting Disk"
+    yes | mkfs "$diskname"
+    log "Mounting $diskname to /data"
+    mkdir /data > /dev/null 2>&1
+    mount "$diskname" /data
+
+    rm -rf /data/*
+
+    testcases=('mixed' '16k_random' '16k_seq' '1mb_random' '1mb_seq' '4k_random' '4k_seq' '4mb_random' '4mb_seq' '512k_random' '512k_seq' '64k_random' '64k_seq' '8mb_random' '8mb_seq' '9mb_random' '9mb_seq')
+    ntests=${#testcases[@]}
+    log "Total Tests: $ntests"
+
+    curdir=$(pwd)
+    cd "$tdir" || exit_with_logs "$FAILED_TEST"
+    for testcase in "${testcases[@]}"; do
+        echo -e "$ctests/$ntests\r"
+        log "Starting $testcase test"
+        startcvt "/dev/sdc" "/data" "$testcase" "$TESTNAME" > "$tdir"/"$testcase".log 2>&1
```

Comment:
don't hard code the disk name, it is changeable during reboot. the /dev/sdc may be os disk